src/code/bank0.asm

```diff
@@ -7090,125 +7130,131 @@ UpdateIndoorRoomStatus::
-    call label_35EE                               ;; 00:3704 $CD $EE $35
-    ld   bc, data_37E4                            ;; 00:3707 $01 $E4 $37
-    ld   de, data_36FC                            ;; 00:370A $11 $FC $36
-    jp   Func_354B                                ;; 00:370D $C3 $4B $35
+    call ObjectPositionToRoomObjectAddress        ;; 00:3704 $CD $EE $35
+    ld   bc, VerticalObjectOffsets                ;; 00:3707 $01 $E4 $37
+    ld   de, OpenDoorLeftObjectIds                ;; 00:370A $11 $FC $36
+    jp   CopyIndoorsMacroObjectsToRoom            ;; 00:370D $C3 $4B $35
 
-data_3710::
-    db $B, $C                                     ;; 00:3710
+OpenDoorRightObjectIds::
+    db $0B, $0C                                   ;; 00:3710
 
 LoadObject_OpenDoorRight::
-    ld   a, $01                                   ;; 00:3712 $3E $01
+    ld   a, ROOM_STATUS_DOOR_OPEN_RIGHT           ;; 00:3712 $3E $01
     call UpdateIndoorRoomStatus                   ;; 00:3714 $CD $C4 $36
     push bc                                       ;; 00:3717 $C5
-    call label_35EE                               ;; 00:3718 $CD $EE $35
-    ld   bc, data_37E4                            ;; 00:371B $01 $E4 $37
-    ld   de, data_3710                            ;; 00:371E $11 $10 $37
-    jp   Func_354B                                ;; 00:3721 $C3 $4B $35
+    call ObjectPositionToRoomObjectAddress        ;; 00:3718 $CD $EE $35
+    ld   bc, VerticalObjectOffsets                ;; 00:371B $01 $E4 $37
+    ld   de, OpenDoorRightObjectIds               ;; 00:371E $11 $10 $37
+    jp   CopyIndoorsMacroObjectsToRoom            ;; 00:3721 $C3 $4B $35
 
-data_3724::
+BossDoorObjectIds::
     db $A4, $A5                                   ;; 00:3724
 
 LoadObject_BossDoor::
     ld   e, $08                                   ;; 00:3726 $1E $08
-    call func_373F                                ;; 00:3728 $CD $3F $37
+    call MakeListOfDoorPositions                  ;; 00:3728 $CD $3F $37
     ; if boss door is not open load door object
     ldh  a, [hRoomStatus]                         ;; 00:372B $F0 $F8
     and  ROOM_STATUS_DOOR_OPEN_UP                 ;; 00:372D $E6 $04
     jp   nz, LoadObject_OpenDoorTop               ;; 00:372F $C2 $B2 $36
     push bc                                       ;; 00:3732 $C5
-    call label_35EE                               ;; 00:3733 $CD $EE $35
-    ld   bc, data_37E1                            ;; 00:3736 $01 $E1 $37
-    ld   de, data_3724                            ;; 00:3739 $11 $24 $37
-    jp   Func_354B                                ;; 00:373C $C3 $4B $35
+    call ObjectPositionToRoomObjectAddress        ;; 00:3733 $CD $EE $35
+    ld   bc, HorizontalObjectOffsets              ;; 00:3736 $01 $E1 $37
+    ld   de, BossDoorObjectIds                    ;; 00:3739 $11 $24 $37
+    jp   CopyIndoorsMacroObjectsToRoom            ;; 00:373C $C3 $4B $35
 
-func_373F::
+; Make lists of the positions of current doors in the room
```

Comment:
Could this comment also explain the function arguments, and where the resulting list is stored?

Reply:
Hello, rabbit hole!